[#157556295] Update the vendored aws-sdk-go to v1.14.4

keymon opened the pull request:
What?
----

We need to use a more modern version in the paas-rds-metrics-collector
As that project uses one module from the rds-broker to discover
the RDS instances, they are coupled in the dependencies of
libraries like this.

Updated running:

  dep ensure -update github.com/aws/aws-sdk-go

Some minor changes were required in the code.

How to test?
-----------

 - Code review
 - tests should pass

Who?
----

Anyone except Mr. Myself

keymon committed:
Update the vendored aws-sdk-go to v1.14.4

We need to use a more modern version in the paas-rds-metrics-collector
As that project uses one module from the rds-broker to discover
the RDS instances, they are coupled in the dependencies of
libraries like this.

Updated running:

  dep ensure -update github.com/aws/aws-sdk-go

Some minor changes were required in the code.

paroxp reviewed (approved):
LGTM. We need to take some of this code into a separate repo, which IMO would make it more manageable.

Waiting for the CI to pass.

keymon committed:
Use reflect.DeepEqual() to compare AWS structs

We have upgraded the aws-go dependency, but now when checking if a
RDS instance has any pending values by comparing the
PendingModifiedValues with an empty struct, it fails because
rds.PendingModifiedValues includes a slice and cannot be
compared with a simple operator anymore.

We use reflect.DeepEqual() instead[1]

[1] https://golang.org/pkg/reflect/#DeepEqual
